src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeLoaderEnvironment.DelegateMarshallerLookup.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+using System.Runtime;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+using Internal.Runtime;
+using Internal.Runtime.Augments;
+using Internal.Runtime.CompilerServices;
+
+using Internal.NativeFormat;
+using Internal.TypeSystem;
+
+namespace Internal.Runtime.TypeLoader
+{
+    public sealed partial class TypeLoaderEnvironment
+    {
+        public unsafe IntPtr TryGetMarshallerForDelegate(RuntimeTypeHandle delegateTypeHandle)
```

Comment:
Doesn't System.Private.Interop have a better home for this? It doesn't seem like this needs access to anything that isn't already publically exposed from `System.Private.TypeLoader`.

Interop will need more tables like this and I don't see why it should set up a camp in the type loader.

Cc @davidwrighton @fadimounir for type loader opinions.